Mathlib/Geometry/Manifold/GroupLieAlgebra.lean

```diff
@@ -0,0 +1,179 @@
+/-
+Copyright (c) 2024 Sébastien Gouëzel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sébastien Gouëzel
+-/
+import Mathlib.Algebra.Lie.Basic
+import Mathlib.Geometry.Manifold.Algebra.LieGroup
+import Mathlib.Geometry.Manifold.VectorField
+
+/-!
+# The Lie algebra of a Lie group
```

Comment:
I believe it's worth explaining the difference between this approach and the one leading up to [LeftInvariantDerivation.instLieAlgebra](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Geometry/Manifold/Algebra/LeftInvariantDerivation.html#LeftInvariantDerivation.instLieAlgebra).

Reply:
I've added a few comments about that.

Reply:
I'm unable to find `instLieAlgebraLieGroupAlgebra` in mathlib. Is that a typo?